records_mover/db/factory.py

```diff
@@ -1,33 +1,38 @@
+# flake8: noqa
+
 from .driver import GenericDBDriver, DBDriver
 import sqlalchemy
-from typing import Union
+from typing import Union, Optional
+from ..check_db_conn_engine import check_db_conn_engine
 
 
-def db_driver(db: Union[sqlalchemy.engine.Engine,
-                        sqlalchemy.engine.Connection],
+def db_driver(db: Optional[Union[sqlalchemy.engine.Engine,
+                           sqlalchemy.engine.Connection]],
+              db_conn: Optional[sqlalchemy.engine.Connection] = None,
+              db_engine: Optional[sqlalchemy.engine.Engine] = None,
               **kwargs) -> DBDriver:
-    engine: sqlalchemy.engine.Engine = db.engine
-    engine_name: str = engine.name
+    db, db_conn, db_engine = check_db_conn_engine(db=db, db_conn=db_conn, db_engine=db_engine)
+    engine_name: str = db_engine.name
```

Comment:
like for instance here, if `db` is a connection not an engine, wouldn't the returned engine be None and this would crash?

Reply:
if db or db_conn is provided, and one of them is a connection, then db_engine would be assigned to db_conn.engine. Relevant lines of code linked:
https://github.com/bluelabsio/records-mover/blob/e5c006015b534cda1477c77ec5ec5da515437619/records_mover/check_db_conn_engine.py#L28-L30

I realize that's not obvious now, so I'm up for rewriting in a way that makes that more clear, maybe naming the functions check_and_assign_db_conn_engine or something like that. Tell me what you think!

Reply:
oh yeah great i missed that but it was what i was envisioning!